pulp_container/app/serializers.py

```diff
@@ -109,8 +108,8 @@ def to_representation(self, value):
         """
         Converts a base_path into a registry path.
         """
-        host = settings.CONTENT_ORIGIN
-        return "".join([host.split("//")[-1], "/", value])
+        request = self.context["request"]
+        return f"{request.get_host()}/{value}"
```

Comment:
:+1:

Reply:
We need to provide two code paths here so that the plugin can stay compatible with older versions of pulpcore that don't pass in this request object. We should file a task to remove the support for the old behavior once we no longer support pulpcore 3.8.

Reply:
After some more discussion on IRC, we decided that it does not make sense to have the two code paths because the bug would still be present until pulpcore 3.9 is released. The other option is to release the pulpcore change as part of 3.8.1. If that does not pose any problems for other plugins, we will merge this PR after 3.8.1 is released.

Reply:
Isn' it the other way around, and this change should bump the minimal supported pulpcore version to 3.8.0-dev?
And therefore only be released after 3.8?

Reply:
@mdellweg that's correct, we would need to raise the pulpcore requirement after 3.8.1 gets released

Reply:
So i think, we need to raise that lower bound to 3.9(.dev?) with this PR. So that attempts to install it from source with the wrong version of pulpcore will fail already.
(we can drop the `dev` in the requirement on release)